File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/RecommendationHistoryModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RecommendationHistoryModel {
    private Long id;
    private LocalDateTime recommendationDate;
    private Boolean accepted;

    private ClientModel client;
    private TravelPackageModel travelPackage;
}

File: frontend/tour-design-web/src/app/models/recommendation-history.model.ts
// src/app/models/recommendation-history.model.ts
import { ClientModel } from './client.model';
import { TravelPackageModel } from './travel-package.model';

export interface RecommendationHistoryModel {
  id: number;
  recommendationDate?: string;
  accepted?: boolean;

  client?: ClientModel;
  travelPackage?: TravelPackageModel;
}
